src/backend/mod.rs

```diff
@@ -0,0 +1,71 @@
+use crate::task::TaskStatus;
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+/// A results [`Backend`] is used to store and retrive the results and status of the tasks.
+#[async_trait]
+pub trait Backend: Send + Sync + Sized {
+    /// The builder type used to create the results backend with a custom configuration.
+    type Builder: BackendBuilder<Backend = Self>;
+
+    /// Update task state and result.
+    async fn store_result(
+        &self,
+        task_id: &str,
+        result: String,
+        state: TaskStatus,
+    ) -> Result<(), ()>;
+
+    /// Get task meta from backend.
+    async fn get_task_meta(&self, task_id: &str) -> ResultMetadata;
+
+    /// Get current state of a given task.
+    async fn get_state(&self, task_id: &str) -> TaskStatus {
+        self.get_task_meta(task_id).await.status
+    }
+
+    /// Get result of a given task.
+    async fn get_result(&self, task_id: &str) -> Option<String> {
```

Comment:
So this will return a serialized form of the result of the task? What happens if the task is still pending?

Reply:
I did this inspired by: https://github.com/celery/celery/blob/7d6c9e4a397f532da5e6c9f72c163af1a9f8cc90/celery/backends/base.py#L472

In this case it should return None. 

Just like in this case. 

https://github.com/celery/celery/blob/7d6c9e4a397f532da5e6c9f72c163af1a9f8cc90/celery/backends/base.py#L871

What do you think?

Reply:
Hmm yea, well this method should probably return a Result either way since there are probably many errors that could happen while fetching